File: common/src/main/java/dev/chasem/cobblemonextras/commands/ItemShout.java
package dev.chasem.cobblemonextras.commands;

import com.cobblemon.mod.common.Cobblemon;
import com.cobblemon.mod.common.api.storage.party.PlayerPartyStore;
import com.cobblemon.mod.common.pokemon.Pokemon;
import com.mojang.brigadier.CommandDispatcher;
import com.mojang.brigadier.arguments.IntegerArgumentType;
import com.mojang.brigadier.context.CommandContext;
import dev.chasem.cobblemonextras.CobblemonExtras;
import dev.chasem.cobblemonextras.permissions.CobblemonExtrasPermissions;
import net.minecraft.item.ItemStack;
import net.minecraft.server.command.ServerCommandSource;
import net.minecraft.server.network.ServerPlayerEntity;
import net.minecraft.text.*;
import net.minecraft.util.Formatting;

import java.util.List;

import static dev.chasem.cobblemonextras.util.PokemonUtility.getHoverText;
import static net.minecraft.server.command.CommandManager.argument;
import static net.minecraft.server.command.CommandManager.literal;

public class ItemShout {

    public void register(CommandDispatcher<ServerCommandSource> dispatcher) {
        dispatcher.register(
                literal("itemshout")
                        .requires(src -> CobblemonExtrasPermissions.checkPermission(src, CobblemonExtras.permissions.ITEMSHOUT_PERMISSION))
                        .executes(this::execute)
        );
    }

    private int execute(CommandContext<ServerCommandSource> ctx) {
        if (ctx.getSource().getPlayer() != null) {
            ServerPlayerEntity player = ctx.getSource().getPlayer();
            ItemStack heldItem = player.getInventory().getMainHandStack();
            HoverEvent.ItemStackContent itemContent = new HoverEvent.ItemStackContent(heldItem);
            MutableText toSend = Text.literal("[").formatted(Formatting.YELLOW)
                    .append(Text.literal("ItemShout").formatted(Formatting.GREEN))
                    .append(Text.literal("] ").formatted(Formatting.YELLOW))
                    .append(player.getDisplayName().copy().append(Text.of(": ")).formatted(Formatting.WHITE));

            Text itemName = heldItem.getName();
            Text hoverAbleName = Texts.join(itemName.getWithStyle(itemName.getStyle()
                    .withHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_ITEM, itemContent))
                    .withUnderline(true)),
                Text.of("")
            );
            toSend.append(hoverAbleName);
            ctx.getSource().getServer().getPlayerManager().getPlayerList().forEach(serverPlayer -> serverPlayer.sendMessage(toSend));
        } else {
            ctx.getSource().sendError(Text.of("Sorry, this is only for players."));
        }
        return 1;
    }


}

File: common/src/main/kotlin/dev/chasem/cobblemonextras/commands/ItemShout.kt
package dev.chasem.cobblemonextras.commands

import com.mojang.brigadier.CommandDispatcher
import com.mojang.brigadier.context.CommandContext
import dev.chasem.cobblemonextras.CobblemonExtras
import dev.chasem.cobblemonextras.permissions.CobblemonExtrasPermissions
import net.minecraft.ChatFormatting
import net.minecraft.commands.CommandSourceStack
import net.minecraft.commands.Commands
import net.minecraft.network.chat.Component
import net.minecraft.network.chat.HoverEvent
import net.minecraft.server.level.ServerPlayer
import net.minecraft.world.item.ItemStack

class ItemShout {
    fun register(dispatcher: CommandDispatcher<CommandSourceStack>) {
        dispatcher.register(
                Commands.literal("itemshout")
                        .requires { src: CommandSourceStack? -> CobblemonExtrasPermissions.checkPermission(src, CobblemonExtras.permissions.ITEMSHOUT_PERMISSION) }
                        .executes { ctx: CommandContext<CommandSourceStack> -> this.execute(ctx) }
        )
    }

    private fun execute(ctx: CommandContext<CommandSourceStack>): Int {
        if (ctx.source.player != null) {
            val player: ServerPlayer = ctx.source.player!!
            val heldItem: ItemStack = player.mainHandItem
//            val itemContent: HoverEvent.ItemStackContent = ItemStackContent(heldItem)
            var itemHover = HoverEvent.ItemStackInfo(heldItem)
            val toSend = Component.literal("[").withStyle(ChatFormatting.YELLOW)
                    .append(Component.literal("ItemShout").withStyle(ChatFormatting.GREEN))
                    .append(Component.literal("] ").withStyle(ChatFormatting.YELLOW))
                    .append(player.displayName!!.copy().append(Component.literal(": ")).withStyle(ChatFormatting.WHITE))

            val itemName = heldItem.displayName
            val hoverAbleName = itemName.copy().withStyle(itemName.style.withHoverEvent(HoverEvent(HoverEvent.Action.SHOW_ITEM, itemHover)).withUnderlined(true))

//            val hoverAbleName = Texts.join(itemName.getWithStyle(itemName.getStyle()
//                    .withHoverEvent(HoverEvent(HoverEvent.Action.SHOW_ITEM, itemContent))
//                    .withUnderline(true)),
//                    Component.literal("")
//            )
            toSend.append(hoverAbleName)
            ctx.source.server.playerList.players.forEach { serverPlayer -> serverPlayer.sendSystemMessage(toSend) }
        } else {
            ctx.source.sendFailure(Component.literal("Sorry, this is only for players."))
        }
        return 1
    }
}
